src/components/Container/ContainerMarginHelper.ts

```diff
@@ -0,0 +1,120 @@
+import { IContainerProps } from './Container';
+
+export type ContainerMarginType = 'xs' | 's' | 'm' | 'l' | 'xl' | 0 | 'none' | string;
+type ReturnValue =  {margin?: string | number} | undefined;
+
+
+export class ContainerMarginHelper {
+
+	public static sizeLookups = new Map<string | number, number>(Object.entries({
+		0: 0,
+		none: 0,
+		xs: 5,
+		s: 10,
+		m: 20,
+		l: 30,
+		xl: 40,
+	}));
+
+	public static getContainerMarginStyle (props: IContainerProps): ReturnValue | undefined {
+		return {
+			...this.processMarginType(props.margin, 'margin'),
+			...this.processMarginType(props.marginTop, 'marginTop'),
+			...this.processMarginType(props.marginRight, 'marginRight'),
+			...this.processMarginType(props.marginBottom, 'marginBottom'),
+			...this.processMarginType(props.marginLeft, 'marginLeft'),
+		};
+	}
+
+	public static processMarginType (marginValue: string | number | undefined, propName: string) {
+		if (marginValue === undefined) {
+			return undefined;
+		}
+
+		return this.wrapReturnValue(this.parseMargin(marginValue), propName);
+	}
+
+	public static parseMargin (margin: string | number): string | number | undefined {
+		const values = (margin as string).split(' ');
+
+		const value = values.reduce((list: any[], value: string) => {
+			list.push(this.formatFinalValue(this.parseExpression(value)));
+			return list;
+		}, []);
+
+		return value.join(' ');
+	}
+
+	public static formatFinalValue (value: string | number | undefined)  {
+		if (value === undefined) {
+			return undefined;
+		}
+
+		return typeof(value) === 'number' && value !== 0 ? `${value}px` : value;
+	}
+
+	public static lookupValue (value: string | number | undefined): number | undefined {
+		if (value === undefined) {
+			return value;
+		}
+
+		return this.sizeLookups.get(value);
+	}
+
+	public static wrapReturnValue (value: string | number | undefined, propName: string): ReturnValue {
+		if (value === undefined) {
+			return undefined;
+		}
+
+		return {
+			[propName]: value,
```

Comment:
See comment above about the `ReturnValue` type.

Reply:
Yeah, I went ahead and added more typings to help prevent missing this going forward 👍